### PR TITLE
Fix race condition on Service process initialisation

### LINKED ISSUE #15: Fix race condition on Service initialisation
https://github.com/belfastelixir/DMS/blob/1efe918ffdce3a1c1e53141f3d7e8eba6f0033a1/lib/dms/service.ex#L25-L35

This would result in a race condition when 2 attempts to start the same service happen at once.  

This should be resolved by leveraging the return value of `DynamicSupervisor.start_child/2` which is the following:
```
  {:ok, pid()}
  | :ignore
  | {:error, {:already_started, pid()} | {:shutdown, term()} | term()}
```
In particular the `:already_started` value sounds like a good option to resolve this: `"If a process with the specified server name already exists, this function returns {:error, {:already_started, pid}} with the PID of that process."`

However the processes are not named, and the Registry is being used, so we may need to explore the `Registry` functionality in order to find the correct solution.

### PR DESCRIPTION
**Pull request by cloggy45**
#15 issue resolved

### CODE REVIEW

**Review by swmcc — APPROVED** [review]
👍 LGTM

File: lib/dms/service.ex [hunk 1]
```diff
@@ -20,20 +20,20 @@ defmodule DMS.Service do
...
+      {:ok, _} ->
+        GenServer.cast(via_registry(id), :ping)
+        :pong
 
-      {:ok, _} = DynamicSupervisor.start_child(@supervisor, child_spec)
-      :pong
+      {:error, {:already_started, _}} ->
+        GenServer.cast(via_registry(id), :ping)
+        :pong
     end
```
Comment (holsee) [inline lib/dms/service.ex:37]: We need to handle the `:pang` / failure condition. Raised issue #19

File: test/dms_test.exs [hunk 2]
```diff
@@ -8,10 +9,24 @@ defmodule DMSTest do
...
+    for resp <- resps do
+      assert :pong = resp
+    end
+  end
+
   test "service should die after 5 seconds" do
     id = "test-svc-2"
     assert :pong = DMS.ping(id)
-    Process.sleep(5_000)
+    Process.sleep(5_010)
```
Comment (holsee) [inline test/dms_test.exs:29]: This is an issue, not only is it slowing the tests but it is fragile and generally a code smell. Raised issue #18

**Review by holsee — COMMENTED** [review]
Raised issues highlighted.

**Pull request merged**